### PR TITLE
CoreDNS user configmap

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/4452

Allow user to configure CoreDNS via a user specific configmap.

### CODE REVIEW

File: service/controller/aws/v8/key/key.go [hunk 1]
```diff
@@ -13,12 +13,13 @@ func ChartSpecs() []key.ChartSpec {
...
 	return []key.ChartSpec{
 		{
-			AppName:       "coredns",
-			ChannelName:   "0-1-stable",
-			ChartName:     "kubernetes-coredns-chart",
-			ConfigMapName: "coredns-values",
-			Namespace:     metav1.NamespaceSystem,
-			ReleaseName:   "coredns",
+			AppName:           "coredns",
+			ChannelName:       "c5bea1db131f72cfee47038fefcd65dece13cc62-basic",
```
Comment (TheoBrigitte) [inline service/controller/aws/v8/key/key.go]: Note: revert this to `0-2-stable` before merging.

File: service/controller/azure/v8/key/key.go [hunk 2]
```diff
@@ -19,12 +19,13 @@ func ChartSpecs() []key.ChartSpec {
...
 		},
 		{
-			AppName:       "coredns",
-			ChannelName:   "0-1-stable",
-			ChartName:     "kubernetes-coredns-chart",
-			ConfigMapName: "coredns-values",
-			Namespace:     metav1.NamespaceSystem,
-			ReleaseName:   "coredns",
+			AppName:           "coredns",
+			ChannelName:       "c5bea1db131f72cfee47038fefcd65dece13cc62-basic",
```
Comment (TheoBrigitte) [inline service/controller/azure/v8/key/key.go]: Note: revert this to `0-2-stable` before merging.

**Comment by TheoBrigitte** [comment]
> We need to then also wire this up in a version in cluster-operator I guess?

What do you mean ? This is already part of aws/v8 and azure/v8. Is there anything else to be done ?

**Comment by kopiczko** [comment]
> This is already part of aws/v8 and azure/v8. Is there anything else to be done ?

No I think @puja108 got confused. This is added to the current WIP version bundle.

**Comment by puja108** [comment]
Yes, sorry, was a bit confused jumping between different things back and forth, all good :)

File: service/controller/aws/v8/key/key.go (hunk 1 above)
Comment (indyfree) [reply service/controller/aws/v8/key/key.go]: FYI (or next time) you can reference `wip-SHA` here. See: https://github.com/giantswarm/kubernetes-coredns/pull/13#issuecomment-431340178

**Comment by TheoBrigitte** [comment]
Successfully tested on ginger

```
$ k get cm coredns-user-values -o yaml
apiVersion: v1
data:
  cache: |
    60 {
          success 524288
          denial 524288
          prefetch 10000
        }
  proxy: |
    foo.com 1.1.1.1
    bar.com 8.8.8.8
kind: ConfigMap
metadata:
  creationTimestamp: 2018-10-26T10:31:27Z
  labels:
    app: coredns
    cluster-operator.giantswarm.io/configmap-type: user
    giantswarm.io/cluster: 0nfux
    giantswarm.io/managed-by: cluster-operator
    giantswarm.io/organization: giantswarm
    giantswarm.io/service-type: managed
  name: coredns-user-values
  namespace: kube-system
  resourceVersion: "12322"
  selfLink: /api/v1/namespaces/kube-system/configmaps/coredns-user-values
  uid: 4b3d686d-d90a-11e8-8d20-025fb2a6a94a

$ k exec -it coredns-868bf64658-frz78 cat /etc/coredns/Corefile
.:53 {
    cache 60 {
  success 524288
  denial 524288
  prefetch 10000
}

    errors
    health
    kubernetes cluster.local 172.31.0.0/16 192.168.0.0/16 {
      fallthrough in-addr.arpa ip6.arpa
      pods insecure
      upstream
    }
    log . {
      class denial
      class error
    }
    prometheus :9153
    proxy foo.com 1.1.1.1
    proxy bar.com 8.8.8.8
    proxy . /etc/resolv.conf
}
```

**Pull request merged**